backend/deploy.js

```diff
@@ -5,15 +5,17 @@ var socketHandler = require('../util/socketHandler.js');
 
 exports.deployPages = function (socket, data) {
   var repoName = data.gitURL.split("/")[4].split(".")[0];
+  var owner = data.gitURL.split("/")[3];
```

Comment:
Define a variable which stores the result of data.gitURL.split('/') and extract reponame and owner from it. No need to do split twice.